### PR TITLE
added packaging pipeline for pre-release

### PR DESCRIPTION
**Pull request by cristianciutea**
This PR will add:
- prerelease workflow for building packages

### CODE REVIEW

File: .github/workflows/ci-prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.19.4'
+          check-latest: true
+
+      - name: Generate distribution sources
+        run: make generate-sources
+
+      # - name: Log into Docker.io
```
Comment (cristianciutea) [inline .github/workflows/ci-prerelease.yml:40]: This should be updated when the repository configuration is ready

File: .goreleaser.yaml [hunk 2]
```diff
@@ -94,3 +95,26 @@ docker_manifests:
       image_templates:
         - newrelic/opentelemetry-collector-releases/nr-otel-collector:latest-amd64
         - newrelic/opentelemetry-collector-releases/nr-otel-collector:latest-arm64
+
+publishers:
```
Comment (rogercoll) [inline .goreleaser.yaml:99]: 👏

File: .github/workflows/scripts/set_version.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+git fetch --prune --unshallow 2> /dev/null || true
+
+tag=$(git describe --tags --abbrev=0)
+
+# Expected tag format <distro>-<version> e.g. distro_name-major.minor.patch
+regex="^(.*)-([0-9]+\.[0-9]+\.[0-9]+)$"
+
+if [[ "${tag}" =~ ${regex} ]]
+then
+    distro="${BASH_REMATCH[1]}"
```
Comment (rubenruizdegauna) [inline .github/workflows/scripts/set_version.sh:16]: TIL BASH_REMATCH :)

**Review by rubenruizdegauna — APPROVED** [review]
LGTM 💪

**Pull request merged**